src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/Contracts/DeploymentOptions.cs

```diff
@@ -241,6 +244,52 @@ public DeploymentOptions()
         }
 
         public DeploymentOptions(DacDeployOptions options)
+        {
+            SetOptions(options);
+        }
+
+        /// <summary>
+        /// create deployment options from the options in a publish profile.xml
+        /// </summary>
+        /// <param name="options">options created from the profile</param>
+        /// <param name="profilePath"></param>
+        public DeploymentOptions(DacDeployOptions options, string profilePath)
+        {
+            // check if defaults need to be set if they aren't specified in the profile
+            string contents = File.ReadAllText(profilePath);
+            if (!contents.Contains("<AllowDropBlockingAssemblies>"))
```

Comment:
Are these all the different settings that might appear in a deploy profile, or might more get added later?  I'm iffy on a series of very similar-looking checks; would it be better/more scalable to loop over a list of tuples?

Reply:
These are the only settings that need to be checked. DacFx and SSMS deploy wizard change these settings from the defaults that DacDeployOptions gets created with, so those defaults also need to get set here if they aren't specified in the profile so that the deployment behavior is consistent.